### ISSUE TITLE
[Feature] Ignore endpoints (SSE) on waitForLoadState(IDLE)

### ISSUE THREAD

**Issue by Nico-DF** [microsoft/playwright#22242]
TL;DR: Maybe having the possibility on the waitForLoadState(IDLE) (on other state it doesn't make sense as it is an event) to specify some pattern to ignore for considering network idle (SSE typically)

I know it might seem a bit weird, but let's say it's half a feature request and half help request on how I could manage it.
My problem is the following:

- I have an action button that will do an HTTP Request
- When this request is over, the angular code will either (depending on some conditions), do a redirect or some other request

My main point here, is that I cannot easily determine what request will follow a click on a specific button.
To solve this, I tried to do the following: (java backend)
```java
click("#myButton");
page.waitForLoadState(LoadState.NETWORKIDLE);
```

And in itself, it works, BUT (that's a big but) my test takes 25s longer (and I have a shitload of tests).

On the surface, all my call are done on the first 2-3 secs, but it wait further 20s...
Then I realised that my app uses SSE: the waitForLoadState(IDLE) will return when "regular" request are done AND the sse session is killed (before being recreated automatically, TTL 30s, etc., classic stuff) I suppose, hence the 20-30 additional seconds if I'm "unlucky"
^I may be wrong here but it does make some kind of sense

So I was wondering if it would be a legit request to ask for something like: 
```java
waitForLoadState(IDLE, new Page.WaitForLoadStateOptions().setIgnorePattern("**/sse**"))
``` 
(or whatever syntax on your desired language)
Or should I just really identify precisely on each of my usecase which specific request/response to wait for?

**Comment by pavelfeldman** [comment]
You should never use NETWORKIDLE for testing, nor you should use IDLE. Steps should always start with a precondition and assertions should always make sure postconditions are met. In Playwright, you would do:

```js
await page.locator('button').click();
await exepect(page.locator('status')).toHaveText('ready')
```

First waits for the button to be actionable, second passes only once the status is 'ready'. You should do the same, rely on Locators and Web Assertions instead of sniffing the network in this case.

**Comment by Nico-DF** [comment]
Ok, thanks for heads up, I'll try dig a bit more on all our usecase (historical app so lot of different path/request depending on workflow) then

Side note, it might be interesting to add your recommendation to waitForLoadState documentation, maybe?
